subsys/sensing/sensor/phy_3d_sensor/phy_3d_sensor.c

```diff
@@ -18,65 +20,222 @@
+		FIELD_GET(GENMASK64(31 + shift, 31), shifted_value) * scale +
+		(FIELD_GET(GENMASK64(30, 0), shifted_value) * scale / BIT(31));
+
+	if (q < 0) {
+		scaled_value = -scaled_value;
+	}
+
+	return scaled_value;
 }
 
-static int phy_3d_sensor_deinit(const struct device *dev)
+static q31_t scaled_int64_to_shifted_q31(int64_t val, int64_t scale,
+		int8_t shift)
 {
-	return 0;
+	return (q31_t)((val * BIT(31 - shift) / scale));
 }
 
-static int phy_3d_sensor_read_sample(const struct device *dev,
-		void *buf, int size)
+static q31_t accel_sensor_value_to_q31(struct sensor_value *val)
 {
-	return 0;
+	int64_t micro_ms2 = sensor_value_to_micro(val);
+	int64_t micro_g = micro_ms2 * 1000000LL / SENSOR_G;
+
+	return scaled_int64_to_shifted_q31(micro_g, 1000000LL,
+			SENSING_ACCEL_Q31_SHIFT);
 }
 
-static int phy_3d_sensor_sensitivity_test(const struct device *dev,
-		int index, uint32_t sensitivity,
-		void *last_sample_buf, int last_sample_size,
-		void *current_sample_buf, int current_sample_size)
+static void accel_q31_to_sensor_value(q31_t q, struct sensor_value *val)
 {
-	return 0;
+	int64_t micro_g = shifted_q31_to_scaled_int64(q,
+			SENSING_ACCEL_Q31_SHIFT, 1000000LL);
+	int64_t micro_ms2 = micro_g * SENSOR_G / 1000000LL;
+
+	sensor_value_from_micro(val, micro_ms2);
 }
 
-static int phy_3d_sensor_set_interval(const struct device *dev, uint32_t value)
+static const struct phy_3d_sensor_custom custom_accel = {
+	.chan_all = SENSOR_CHAN_ACCEL_XYZ,
+	.shift = SENSING_ACCEL_Q31_SHIFT,
+	.q31_to_sensor_value = accel_q31_to_sensor_value,
+	.sensor_value_to_q31 = accel_sensor_value_to_q31,
+};
+
+static q31_t gyro_sensor_value_to_q31(struct sensor_value *val)
 {
-	return 0;
+	int64_t micro_rad = (int64_t)sensor_value_to_micro(val);
+	int64_t micro_deg = micro_rad * 180000000LL / SENSOR_PI;
+
+	return scaled_int64_to_shifted_q31(micro_deg, 1000000LL,
+			SENSING_GYRO_Q31_SHIFT);
 }
 
-static int phy_3d_sensor_get_interval(const struct device *dev,
-		uint32_t *value)
+static void gyro_q31_to_sensor_value(q31_t q, struct sensor_value *val)
 {
-	return 0;
+	int64_t micro_deg = shifted_q31_to_scaled_int64(q,
+			SENSING_GYRO_Q31_SHIFT, 1000000LL);
+	int64_t micro_rad = micro_deg * SENSOR_PI / 180000000LL;
+
+	sensor_value_from_micro(val, micro_rad);
 }
 
-static int phy_3d_sensor_set_sensitivity(const struct device *dev,
-		int index, uint32_t value)
+static const struct phy_3d_sensor_custom custom_gyro = {
+	.chan_all = SENSOR_CHAN_GYRO_XYZ,
+	.shift = SENSING_GYRO_Q31_SHIFT,
+	.q31_to_sensor_value = gyro_q31_to_sensor_value,
+	.sensor_value_to_q31 = gyro_sensor_value_to_q31,
+};
+
+static int phy_3d_sensor_init(const struct device *dev)
 {
+	const struct phy_3d_sensor_config *cfg = dev->config;
+	struct phy_3d_sensor_data *data = dev->data;
+
+	switch (cfg->sensor_type) {
+	case SENSING_SENSOR_TYPE_MOTION_ACCELEROMETER_3D:
+		data->custom = &custom_accel;
+		break;
+	case SENSING_SENSOR_TYPE_MOTION_GYROMETER_3D:
+		data->custom = &custom_gyro;
+		break;
```

Comment:
From what I can tell this is still the same as my previous comments. You're creating 1 `struct device` per type. You need to have 1 `struct device` per physical sensor and _N_ `struct sensor_info` objects based on the number of types that are supported.